src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeParse.cs

```diff
@@ -5998,9 +5999,10 @@ struct DateTimeToken
     //
     // The buffer to store temporary parsing information.
     //
-    internal unsafe struct DateTimeRawInfo
+    [StructLayout(LayoutKind.Auto)]
```

Comment:
What is the goal of the `Auto` in this case? Are we seeing some performance benefit here?

Reply:
Hmmm. This seems likely because the `Auto` algorithm can reorder field layout. I would suggest adding a comment here as to why this is being done. A valuetype with `Auto` layout always makes the interop team ask follow-ups.